lxc/image.go

```diff
@@ -286,3 +307,90 @@ func (c *imageCmd) run(config *lxd.Config, args []string) error {
 		return fmt.Errorf(gettext.Gettext("Unknown image command %s"), args[0])
 	}
 }
+
+func dereferenceAlias(d *lxd.Client, inName string) string {
+	imageList, err := d.ListImages()
+	if err != nil {
+		return ""
+	}
+	inLen := len(inName)
+	for _, urln := range imageList {
+		prefix := "/1.0/images/"
+		offset := len(prefix)
+		if len(urln) < offset+1 {
+			continue
+		}
+		n := urln[offset:]
+		if len(n) < inLen {
+			continue
+		}
+		if n[:inLen] == inName {
+			return n
```

Comment:
Can we just do a sort of the list and take the shortest one here, so we don't need all this code?